web-app/src/utils/validations.js

```diff
@@ -10,11 +12,20 @@ const isIntervalValid = (low, high) =>
 const isTimeIntervalValid = (low, high) =>
   (low == null || high == null || low <= high) && low !== '' && high !== ''
 
+const isCurrencyValid = (currency, codeLists) =>
+  currency == null || !codeLists || codeLists.ISO4217.codes[currency]
```

Comment:
how can currency be valid if `codeLists` are null?
maybe the condition should look like `currency == null || currency in codeLists.ISO4217.codes`